### ISSUE TITLE
422 Unprocessable Entity when creating GKE cluster

### ISSUE THREAD

**Issue by daanemanz** [rancher/terraform-provider-rancher2-archive#34]
When creating a GKE cluster with the `rancher2_cluster` resource, I specify `credential = "${file("credentials.json")"` in the `gke_config` block. However, I'm receiving the following error:
```
* rancher2_cluster.ci-cluster: Bad response statusCode [422]. Status [422 Unprocessable Entity]. Body: [message=credential=MissingRequired 422: , baseType=error, code=InvalidFormat, fieldName=googleKubernetesEngineConfig] from [https://rancher.domain.tld/v3/clusters]
```

Even when the credentials in `credentials.json` can be used successfully in the Rancher UI to the same end.

**Comment by daanemanz** [comment]
Additional info: `terraform plan` shows this in the output:
```
Terraform will perform the following actions:

  + rancher2_cluster.ci-cluster
      id:                                             <computed>
      annotations.%:                                  <computed>
      cluster_registration_token.#:                   <computed>
      description:                                    "Test Cluster"
      gke_config.#:                                   "1"
      gke_config.0.credential:                        "{\n  \"type\": \"service_account\",\n  \"project_id\": \"<redacted>\",\n  \"private_key_id\": \"<redacted>\",\n  \"private_key\": \"-----BEGIN PRIVATE KEY-----\\n<redacted>\\n-----END PRIVATE KEY-----\\n\",\n  \"client_email\": \"<redacted>\",\n  \"client_id\": \"<redacted>\",\n  \"auth_uri\": \"https://accounts.google.com/o/oauth2/auth\",\n  \"token_uri\": \"https://oauth2.googleapis.com/token\",\n  \"auth_provider_x509_cert_url\": \"https://www.googleapis.com/oauth2/v1/certs\",\n  \"client_x509_cert_url\": \"https://www.googleapis.com/robot/v1/metadata/x509/terraform%40<redacted>\"\n}\n"
      gke_config.0.disk_size_gb:                      "100"
      gke_config.0.enable_alpha_feature:              "false"
      gke_config.0.enable_horizontal_pod_autoscaling: "true"
      gke_config.0.enable_http_load_balancing:        "true"
      gke_config.0.enable_kubernetes_dashboard:       "false"
      gke_config.0.enable_legacy_abac:                "false"
      gke_config.0.enable_network_policy_config:      "false"
      gke_config.0.enable_stackdriver_logging:        "true"
      gke_config.0.enable_stackdriver_monitoring:     "true"
      gke_config.0.labels.%:                          <computed>
      gke_config.0.machine_type:                      "g1-small"
      gke_config.0.master_version:                    "v1.11.7-gke.6"
      gke_config.0.network:                           "rancher"
      gke_config.0.node_count:                        "3"
      gke_config.0.project_id:                        "<redacted>"
      gke_config.0.sub_network:                       "rancher"
      gke_config.0.zone:                              "us-central1-a"
      kind:                                           "gke"
      labels.%:                                       <computed>
      name:                                           "ci-cluster"
```

**Comment by rawmind0** [comment]
@daanemanz, issue is caused by a typo in the gke code. It's solved with PR  https://github.com/rancher/terraform-provider-rancher2/pull/35 
Could you please rebuild terraform provider and test it??

**Comment by daanemanz** [comment]
@rawmind0 I haven't had a chance to test it but the provider built for me. I'll let you know!

**Comment by rawmind0** [comment]
@daanemanz, have you had a chance to test the fix??

**Comment by daanemanz** [comment]
@rawmind0 yes, I have, seems to work now! However, another error occurred which is filed here https://github.com/rancher/rancher/issues/18511

**Comment by rawmind0** [comment]
@daanemanz, thanks for the feedback...Please, reopen issue if needed...

BTW, I'm going to update issue rancher/rancher#18511, that was solved with PR #37...
